DPGAnalysis/MuonSysAging/plugins/CSCChamberMasker.cc

```diff
@@ -0,0 +1,333 @@
+                  } else {
+                      // std::cout << "We're !!dropping!! this digi in " << cscDetId << std::endl;
+                  }
+              } 
+          }
+      }
+  }
+
+  // Handle wire digis
+  if(!m_wireDigiToken.isUninitialized())
+  {
+
+      edm::Handle<CSCWireDigiCollection> cscWireDigis;
+      event.getByToken(m_wireDigiToken, cscWireDigis);
+
+      for (CSCWireDigiCollection::DigiRangeIterator j=cscWireDigis->begin(); j!=cscWireDigis->end(); j++) {
+          std::vector<CSCWireDigi>::const_iterator digiItr = (*j).second.first;
+          std::vector<CSCWireDigi>::const_iterator last = (*j).second.second;
+
+          CSCDetId const cscDetId=(*j).first;
+
+          CSCDetId chId = CSCDetId(cscDetId.endcap(), cscDetId.station(), cscDetId.ring(), cscDetId.chamber(), 0);
+
+          for( ; digiItr != last; ++digiItr) {
+              auto chEffIt = m_CSCEffs.find(chId);
+              if (chEffIt != m_CSCEffs.end()) {
+                  // std::cout<<"  Wire Digi in chamber of interest: " << chId <<std::endl;
+                  std::pair<unsigned int, float> typeEff = chEffIt->second;
+                  int type = typeEff.first % 10;
+                  int layer = typeEff.first/10;
+                  // std::cout << " type: " << type << " layer: " << layer << std::endl;
+                  bool doRandomize = false;
+                  // if ((type == EFF_WIRES || type == EFF_CHAMBER) && (layer == 0 || cscDetId.layer() == layer) && randGen.flat() <= typeEff.second) {
+                  if ((type == EFF_WIRES || type == EFF_CHAMBER) && (layer == 0 || cscDetId.layer() == layer)) doRandomize = true;
+                  // if ((type == EFF_WIRES || type == EFF_CHAMBER) && (layer == 0 || cscDetId.layer() == layer) && randGen.flat() <= typeEff.second) {
+                  if (!doRandomize  || (randGen.flat() <= typeEff.second)) {
+                      // std::cout << "We're keeping this wire digi in " << cscDetId << std::endl;
+                      filteredWireDigis->insertDigi(cscDetId,*digiItr);
+                  } else {
+                      // std::cout << "We're !!dropping!! this digi in " << cscDetId << std::endl;
+                  }
+              } 
+          }
+      }
+  }
+
+  // Don't touch CLCT or ALCT digis
+  if(!m_clctDigiToken.isUninitialized())
+  {
+      edm::Handle<CSCCLCTDigiCollection> cscCLCTDigis;
+      event.getByToken(m_clctDigiToken, cscCLCTDigis);
+      for (CSCCLCTDigiCollection::DigiRangeIterator j=cscCLCTDigis->begin(); j!=cscCLCTDigis->end(); j++) {
+          std::vector<CSCCLCTDigi>::const_iterator digiItr = (*j).second.first;
+          std::vector<CSCCLCTDigi>::const_iterator last = (*j).second.second;
+          CSCDetId const cscDetId=(*j).first;
+          for( ; digiItr != last; ++digiItr) {
+              filteredCLCTDigis->insertDigi(cscDetId,*digiItr);
+          }
+      }
+  }
+
+  if(!m_alctDigiToken.isUninitialized())
```

Comment:
a template member function was added to merge almost identical code, please have a look.